pkg/virt-api/rest/lifecycle.go

```diff
@@ -210,17 +214,21 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 	}
 
 	var oldGracePeriodSeconds int64
-	patchType := types.MergePatchType
 	var patchErr error
 	if hasVMI && !vmi.IsFinal() && bodyStruct.GracePeriod != nil {
+		patchSet := patch.New()
 		// used for stopping a VM with RunStrategyHalted
 		if vmi.Spec.TerminationGracePeriodSeconds != nil {
 			oldGracePeriodSeconds = *vmi.Spec.TerminationGracePeriodSeconds
+			patchSet.AddOption(patch.WithTest("/spec/terminationGracePeriodSeconds", *vmi.Spec.TerminationGracePeriodSeconds))
+		} else {
+			patchSet.AddOption(patch.WithTest("/spec/terminationGracePeriodSeconds", nil))
 		}
 
-		bodyString := getUpdateTerminatingSecondsGracePeriod(*bodyStruct.GracePeriod)
-		log.Log.Object(vmi).V(2).Infof("Patching VMI: %s", bodyString)
-		_, err = app.virtCli.VirtualMachineInstance(namespace).Patch(context.Background(), vmi.GetName(), patchType, []byte(bodyString), metav1.PatchOptions{DryRun: bodyStruct.DryRun})
+		patchSet.AddOption(patch.WithReplace("/spec/terminationGracePeriodSeconds", *bodyStruct.GracePeriod))
+		patchBytes, err := patchSet.GeneratePayload()
+		log.Log.Object(vmi).V(2).Infof("Patching VMI: %s", string(patchBytes))
```

Comment:
there is the error handling missing here. Either you ignore it or you should handle it

Reply:
Nice catch!